sys/log/full/src/log_cbmem.c

```diff
@@ -70,27 +73,29 @@ log_cbmem_append_mbuf_body(struct log *log, const struct log_entry_hdr *hdr,
 {
     struct cbmem *cbmem;
     struct cbmem_scat_gath sg = {
-        .entries = (struct cbmem_scat_gath_entry[3]) {
+        .entries = (struct cbmem_scat_gath_entry[]) {
             {
                 .flat_buf = hdr,
                 .flat_len = LOG_BASE_ENTRY_HDR_SIZE,
             },
+#if MYNEWT_VAL(LOG_VERSION) > 2
             {
                 .flat_buf = hdr->ue_imghash,
                 .flat_len = 0,
             },
+#endif
             {
                 .om = om,
             },
         },
         .count = 3,
```

Comment:
Here too. count is only 2 if LOG_VERSION >= 3

Reply:
Did you mean count is only 2 if LOG_VERSION < 3 ?

Reply:
Indeed :)

Reply:
Done.